arelle/ValidateXbrlCalcs.py

```diff
@@ -468,6 +565,12 @@ def roundValue(value, precision=None, decimals=None, scale=None):
     return vRounded
 
 def rangeValue(value, decimals=None):
+    if isinstance(value, list):
+        if len(value) == 1 and isinstance(value[0], ModelFact):
+            return rangeValue(value[0].xValue, inferredDecimals(value[0]))
+        # return intersection of fact value ranges
+    if isinstance(value, ModelFact):
```

Comment:
This can cause an exception.

`rangeValue` is used by ESEF validation (and possibly other places).  If it is called before the `validate` function above, `ModelFact` doesn't get initialised and so is `None`, which causes `isinstance` to throw an exception.

I assume that `ModelFact` isn't imported directly due to a circular dependency, but I've not chased it down.

Reply:
This appears to be the only use of `ModelFact` in this file, so moving the import into this function would be one fix.

I done think there's any benefit to using the global + conditional import compared to just putting the import inside a function: Python won't import the same file twice.

Reply:
somewhere at sometime it was in the list of bad and inefficient python practices to import in functions